src/Http/Http.php

```diff
@@ -339,101 +315,44 @@ public static function setMode(string $value): void
 
     /**
-     * Get allow override
+     * Get Container
      *
-     *
-     * @return bool
+     * @return Container
      */
-    public static function getAllowOverride(): bool
+    public function getContainer(): Container
     {
-        return Router::getAllowOverride();
+        return $this->container;
     }
 
     /**
-     * Set Allow override
-     *
-     *
-     * @param  bool  $value
-     * @return void
+     * Set Container
      */
-    public static function setAllowOverride(bool $value): void
+    public function setContainer(Container $container): void
     {
-        Router::setAllowOverride($value);
+        $this->container = $container;
     }
 
     /**
-     * If a resource has been created return it, otherwise create it and then return it
+     * Get allow override
      *
-     * @param  string  $name
-     * @param  bool  $fresh
-     * @return mixed
      *
-     * @throws Exception
+     * @return bool
      */
-    public function getResource(string $name, string $context = 'utopia', bool $fresh = false): mixed
+    public static function getAllowOverride(): bool
     {
-        if ($name === 'utopia') {
-            return $this;
-        }
-
-        $this->resources[$context] ??= [];
-
-        $resourcesCallback = &self::$resourcesCallbacks[$context] ?? [];
-        if(empty($resourcesCallback) || !\array_key_exists($name, $resourcesCallback)) {
-            $resourcesCallback = &self::$resourcesCallbacks['utopia'];
-        }
-
-        if (!\array_key_exists($name, $this->resources[$context]) || $fresh || ($resourcesCallback[$name]['reset'][$context] ?? true)) {
-            if (!\array_key_exists($name, $resourcesCallback)) {
-                throw new Exception('Failed to find resource: "' . $name . '"');
-            }
-
-            $this->resources[$context][$name] = \call_user_func_array(
-                $resourcesCallback[$name]['callback'],
-                $this->getResources($resourcesCallback[$name]['injections'], $context)
-            );
-        }
-
-        $resourcesCallback[$name]['reset'][$context] = false;
-        return $this->resources[$context][$name];
+        return Router::getAllowOverride();
     }
 
     /**
-     * Get Resources By List
+     * Set Allow override
      *
-     * @param  array  $list
-     * @return array
-     */
-    public function getResources(array $list, string $context = 'utopia'): array
-    {
-        $resources = [];
-
-        foreach ($list as $name) {
-            $resources[$name] = $this->getResource($name, $context);
-        }
-
-        return $resources;
-    }
-
-    /**
-     * Set a new resource callback
      *
-     * @param  string  $name
-     * @param  callable  $callback
-     * @param  array  $injections
+     * @param  bool  $value
      * @return void
-     *
-     * @throws Exception
      */
-    public static function setResource(string $name, callable $callback, array $injections = [], string $context = 'utopia'): void
```

Comment:
Why not also provide the setResource so that people can still set dependency using this method, but internally use the container to set these dependencies? I still like having this way to set resource here?

Reply:
You can now do getContainer->set();